framework/src/problems/FEProblemBase.C

```diff
@@ -5926,10 +5926,13 @@ FEProblemBase::solve(const unsigned int nl_sys_num)
   _fail_next_nonlinear_convergence_check = false;
 
   if (_solve)
+  {
     _current_nl_sys->solve();
-
-  if (_solve)
     _current_nl_sys->update();
+  }
+  else
+    // If we are not solving, we do not care about seeing unused petsc options
+    Moose::PetscSupport::setSinglePetscOption("-options_left", "0");
```

Comment:
do we need to do this at the last minute during the solve?
I d only do that if the options get cleared but I dont think they do?

Reply:
I tried putting this at the end of the constructor, but it didn't work. Do you know what MOOSE's interface is for clearing options? If so, I can set a checkpoint to see if that is getting hit.

Reply:
I dont know. 
But you want to set this after the other PETSc options have been added maybe?
so somewhat after the constructor

Reply:
Figured it out. See changes and explanatory comments.